adagrams.rb

```diff
@@ -0,0 +1,148 @@
+require 'pry'
+
+letter_pool = %w(a a a a a a a a a b b c c d d d d e e e e e e e e e e e e f f g g g h h i i i i i i i i i j k l l l l m m n n n n n n o o o o o o o o p p q r r r r r r s s s s t t t t t t u u u u v v w w x y y z)
+
+
+
+# Wave one method
+def draw_letters
+  remaining_pool = %w(a a a a a a a a a b b c c d d d d e e e e e e e e e e e e f f g g g h h i i i i i i i i i j k l l l l m m n n n n n n o o o o o o o o p p q r r r r r r s s s s t t t t t t u u u u v v w w x y y z)
+  remaining_pool.map!(&:upcase)
+  letters_in_hand = []
+  
+  10.times do |letter|
+    letter = remaining_pool.sample
+    letters_in_hand.push(letter)
+    remaining_pool.slice!(remaining_pool.index(letter))
+  end 
+  # p remaining_pool
+  return letters_in_hand
+end 
+
+
+
+# Wave 2: uses_available_letters?
+# Need to review lines 57-67, b/c failing test for returning true if the submitted letters are valid against the drawn letters
+def uses_available_letters?(input, letters_in_hand)
+  
+  # Make input word uppercase and then split it into an array
+  input_array = input.upcase.chars
+  
+  # Check if the input string's characters exist in the user hand 
+  
+  input_array.each do |letter|
+    unless letters_in_hand.include?(letter) 
+      return false
+    end
+  end 
+  
+  # Check if the input string's characters do not exceed the amount of characters 
+  # available in letters_in_hand
+  # Making a hash from letters_in_hand with letters as keys, and counts as values
+  letters_hand_hash = {} 
+  letters_in_hand.each do |letter| #looping thru hand, counting instances of each character
+    
+    found_letter = letter
+    if letters_hand_hash.key?(found_letter)
+      letters_hand_hash[found_letter] += 1
+    else
+      letters_hand_hash[found_letter] = 1
+    end 
+  end 
+  
+  # making a hash from user_input with letters as keys and counts as values 
+  input_hash = {}
+  input_array.each do |letter|
```

Comment:
I think making a second hash complicates this a lot. What if you just used the input string to count down the number of letters from `letters_hand_hash`?